src/ui/component/textField/textField.tsx

```diff
@@ -0,0 +1,98 @@
+import { type FC, type ChangeEvent, useCallback } from 'react'
+import classNames from 'classnames'
+import { Icon } from '@/ui/component/icon/icon'
+import './textField.scss'
+
+type Icons = {
+  startIcon?: JSX.Element
+  endIcon?: JSX.Element
+}
+
+type TextFieldProps = {
+  id: string
+  label: string
+  onChange: (value: string) => void
+  value?: string
+  error?: string
+  required?: boolean
+  disabled?: boolean
+  readonly?: boolean
+  multiline?: boolean
+  resizable?: boolean
+  icons?: Icons
+}
+
+// eslint-disable-next-line max-lines-per-function
+export const TextField: FC<TextFieldProps> = ({
+  id,
+  label,
+  onChange,
+  value,
+  error,
+  readonly = false,
+  required = false,
+  disabled = false,
+  multiline = false,
+  resizable = false,
+  icons
+}) => {
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
+      onChange(e.target.value)
+    },
+    [onChange]
+  )
+
+  const inputClassName = classNames(
+    'okp4-dataverse-portal-text-field',
+    { filled: value },
+    { error },
+    { resizable },
+    { 'with-start-icon': icons?.startIcon },
+    { 'with-end-icon': icons?.endIcon }
+  )
+
+  return (
+    <div className="okp4-dataverse-portal-text-field-main">
+      {multiline ? (
+        <textarea
+          className={inputClassName}
+          disabled={disabled}
+          id={id}
+          name={id}
+          onChange={handleChange}
+          readOnly={readonly}
+          required={required}
+          value={value}
```

Comment:
You can prevent from duplicating the same props between `textarea` and `input` by using `React.cloneElement`

Reply:
Didn't find a way to prevent duplication using `cloneElement` since these elements represents different html tags. Instead, the duplicated elements attributes has been set in an object that is spread at JSX level to prevent duplication.

Is this an appropriate solution ?